### ISSUE TITLE
sql: surface errors for asynchronous schema changes

### ISSUE THREAD

**Issue by vivekmenezes** [cockroachdb/cockroach#2985]
Some errors can only be detected while applying a schema change; e.g. creating a unique constraint on a column that is not unique. These errors should be surfaced back to the goroutine that is waiting on the completion of the schema change.

**Comment by tamird** [comment]
Closing in favour of #4646.
